src/main/java/com/github/tombentley/krpccodegen/main/KrpcGenerator.java

```diff
@@ -44,21 +45,68 @@
 import com.github.tombentley.krpccodegen.model.KrpcSchemaObjectWrapper;
 import com.github.tombentley.krpccodegen.model.SnakeCase;
 import com.github.tombentley.krpccodegen.schema.MessageSpec;
-import com.github.tombentley.krpccodegen.schema.MessageSpecType;
 import com.github.tombentley.krpccodegen.schema.StructRegistry;
+
 import freemarker.template.Configuration;
 import freemarker.template.TemplateException;
 import freemarker.template.TemplateExceptionHandler;
 import freemarker.template.Version;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
 
 public class KrpcGenerator {
-    private static final Logger LOGGER = LoggerFactory.getLogger(KrpcGenerator.class);
 
-    static final String JSON_SUFFIX = ".json";
+    public static class Builder {
+
+        private Logger logger;
+
+        private File schemaDir;
+        private String schemaFilter;
+
+        private File templateDir;
+        private List<String> templateNames;
+
+        private File outputDir;
+        private String outputFilePattern;
+
+
+        public Builder withLogger(Logger logger) {
+            this.logger = logger;
+            return this;
+        }
+
+        public Builder withSchemaDir(File schemaDir) {
```

Comment:
I wonder whether 'schema' is the right term to use. I used it in `KrpcGenerator`, but the code I copied and modified from Kafka uses `MessageSpec` for the same concept, and in hindsight I realise that it's confusing having two names for the same thing. On the one hand Schema is shorter, obviously. But OTOH when copying `MessageSpec` and those classes from Kafka I tried to minimise the changes I made, so I don't think renaming `MessageSpec` → `Schema` would be the right move.